### ISSUE TITLE
Feature filtering

### ISSUE THREAD

**Issue by bhousel** [facebookincubator/RapiD#449]
The old feature classification code was removed for the v2 prototype, but we should put something back in that is like that or better.  

When I wrote that code originally, it was mostly designed to very quickly cull objects from the scene because SVG could not draw them all.  Users will still want to manage the complexity of the scene (e.g. hide landuse) but we need this now more for usability, not render performance.

**Comment by tordans** [comment]
A general +1. And maybe now, that performance is not the issue anymore, the feature could be more about disabling features (grayed on, unclickable) than actually hiding them. [I just wrote a bit more about this in the other issue.](https://github.com/facebookincubator/RapiD/issues/451#issuecomment-1101608377). At least for me, the usecase was mostly less about performance, but more about a way to focus my editing and getting rid of stuff that I would only break by accident, like borders, underground stuff, … – However, I find myself toggling filters because I need the objects for context.

**Comment by bhousel** [comment]
Yes, I'd love to be able to build something new now that performance is less of a concern.

One thing I always wished we could have included before was an ability to let people filter features at different levels - underground, aboveground, or work at different levels..  This was a thing that never [made it into iD](https://github.com/openstreetmap/iD/issues/4780) but did exist in other forks or prototypes.

JOSM also has a pretty advanced filtering capability that lets you filter by tag or regex.  So there are lots of good examples to draw inspiration from..

**Comment by tsmock** [comment]
I don't know if you want to reuse the JOSM syntax, but simonpoole/Woazboat did some documenting here: https://github.com/simonpoole/JosmFilterDoc .

**Comment by bhousel** [comment]
We have a bunch to do to improve the filtering and performance and there are other issues that relate to this one too, but as a first step, I restored this little status chip that we previously had in Rapid v1.  

I think it's important to let users know at a glance whether any of the filters are actively hiding some things.
![filter status](https://github.com/user-attachments/assets/aa6891c1-27eb-4072-8197-98bc624b9d24)
